dln/operator.py

```diff
@@ -250,66 +230,124 @@ def _mini_batch(self, inputs, batch_size=20):
             input_batch = inputs[batch_size * i : batch_size * (i + 1)]
             yield input_batch
 
-    def generate(self, inputs, async_generation=True, batch_size=20, **kwargs):
-        if type(inputs) is not list:
+    def generate(
+        self,
+        inputs: Union[List[str], str],
+        async_generation: bool = True,
+        batch_size: int = 20,
+        **kwargs,
+    ) -> List[str]:
+        if not isinstance(inputs, list):
             inputs = [inputs]
-
-        kwargs.pop("output_space", None)
         generation_options = self.generation_options.copy()
         generation_options.update(**kwargs)
 
-        if self.engine in ("gpt-3.5-turbo", "gpt-35-turbo", "gpt-4", "gpt-4-32k", "gpt-4-0613", "any"):
+        if self.engine in self.CHAT_COMPLETION_MODELS:
             if "return_logprobs" in generation_options:
+                logging.warn(
+                    "return_logprobs is not supported for chat completion models"
+                )
                 del generation_options["return_logprobs"]
 
             if async_generation is True:
                 # async call api, devide to mini batches to avoid call rate limit
                 outputs = []
                 for input_batch in self._mini_batch(inputs, batch_size=10):
                     outputs_batch = asyncio.run(
-                        self.gather_chat_response(input_batch, **generation_options)
+                        self._gather_chat_response(input_batch, **generation_options)
                     )
                     outputs = outputs + outputs_batch
             else:
                 # call api one by one
                 outputs = [
-                    self.get_chat_completion_response(_input, **generation_options)
+                    asyncio.run(
+                        self._aget_chat_completion_response(_input, **generation_options)
+                    )
                     for _input in inputs
                 ]
         else:
-            # devide to mini batches (max batch size = 20 according to openai)
+            # completion_models, devide to mini batches (max batch size = 20 according to openai)
             outputs = []
             for input_batch in self._mini_batch(inputs, batch_size=batch_size):
-                outputs_batch = self.get_completion_response(
+                outputs_batch = self._get_completion_response(
                     input_batch, **generation_options
                 )
                 outputs = outputs + outputs_batch
         return outputs
 
 
-def forward_instantiate(model_name="text-davinci-003", **generation_options):
-    global forward_interpreter
+class VLLM(LLM):
 
-    if forward_interpreter is None:
-        forward_interpreter = GPT(model_name, **generation_options)
-    else:
-        print("Forward interpreter already instantiated.")
-        pass
+    def __init__(self, model_name="text-davinci-003", **generation_options):
```

Comment:
Should the default model be something else than text-davinci-003?

Reply:
It needs to match the one from the running vLLM instance, right? Maybe that shouldn't be optional then.

Reply:
I see that's what you did :)